docs/integration_pipeline_controller.md

```diff
@@ -0,0 +1,37 @@
+<div align="center"><h1>IntegrationPipeline Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+
+   %% Node definitions
+  predicate((PREDICATE: <br>integration pipeline <br> reconciliation))
+  get_pipeline_run[Start pipeline run]
+  get_component[Get component]
+  get_application[Get application]
```

Comment:
This can probably be rolled into the same node with get_component and maybe even with get_pipeline_run.